src/main/java/be/acara/frontend/util/ExtractYoutubeId.java

```diff
@@ -0,0 +1,34 @@
+package be.acara.frontend.util;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ExtractYoutubeId {
+    final String youTubeUrlRegEx = "^(https?)?(://)?(www.)?(m.)?((youtube.com)|(youtu.be))/";
+    final String[] videoIdRegex = { "\\?vi?=([^&]*)","watch\\?.*v=([^&]*)", "(?:embed|vi?)/([^/?]*)", "^([A-Za-z0-9\\-]*)"};
+
+    public String extractVideoIdFromUrl(String url) {
+        String youTubeLinkWithoutProtocolAndDomain = youTubeLinkWithoutProtocolAndDomain(url);
+
+        for(String regex : videoIdRegex) {
```

Comment:
Stream is more elegant

Reply:
Not necessarily. Why would you use a stream in this case?